Assets/Mirror/Components/NetworkTransform/NetworkTransformHybrid2022.cs

```diff
@@ -0,0 +1,1131 @@
+// Quake: https://www.jfedor.org/quake3/
+//
+// Base class for NetworkTransform and NetworkTransformChild.
+// => simple unreliable sync without any interpolation for now.
+// => which means we don't need teleport detection either
+//
+// several functions are virtual in case someone needs to modify a part.
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+
+namespace Mirror
+{
+    [AddComponentMenu("Network/Network Transform Hybrid")]
+    public class NetworkTransformHybrid2022 : NetworkBehaviour
+    {
+        // target transform to sync. can be on a child.
+        [Header("Target")]
+        [Tooltip("The Transform component to sync. May be on this GameObject, or on a child.")]
+        public Transform target;
+
+        // Is this a client with authority over this transform?
+        // This component could be on the player object or any object that has been assigned authority to this client.
+        protected bool IsClientWithAuthority => isClient && authority;
+
+        [Tooltip("Buffer size limit to avoid ever growing list memory consumption attacks.")]
+        public int bufferSizeLimit = 64;
+        internal SortedList<double, TransformSnapshot> clientSnapshots = new SortedList<double, TransformSnapshot>();
+        internal SortedList<double, TransformSnapshot> serverSnapshots = new SortedList<double, TransformSnapshot>();
+
+        // CUSTOM CHANGE: bring back sendRate. this will probably be ported to Mirror.
+        // TODO but use built in syncInterval instead of the extra field here!
+        [Header("Synchronization")]
+        [Tooltip("Send N snapshots per second. Multiples of frame rate make sense.")]
+        public int sendRate = 30; // in Hz. easier to work with as int for EMA. easier to display '30' than '0.333333333'
+        public float sendInterval => 1f / sendRate;
+        // END CUSTOM CHANGE
+
+        [Tooltip("Occasionally send a full reliable state to delta compress against. This only applies to Components with SyncMethod=Unreliable.")]
+        public int baselineRate = 1;
+        public float baselineInterval => baselineRate < int.MaxValue ? 1f / baselineRate : 0; // for 1 Hz, that's 1000ms
+        double lastBaselineTime;
+        double lastDeltaTime;
+
+        // delta compression needs to remember 'last' to compress against.
+        // this is from reliable full state serializations, not from last
+        // unreliable delta since that isn't guaranteed to be delivered.
+        byte lastSerializedBaselineTick = 0;
+        Vector3 lastSerializedBaselinePosition = Vector3.zero;
+        Quaternion lastSerializedBaselineRotation = Quaternion.identity;
+
+        // save last deserialized baseline to delta decompress against
+        byte lastDeserializedBaselineTick = 0;
+        Vector3 lastDeserializedBaselinePosition = Vector3.zero;
+        Quaternion lastDeserializedBaselineRotation = Quaternion.identity;
```

Comment:
"assigned, but never used"

Reply:
yep, needed for delta compression. gonna put comments on there